src/main/java/com/getindata/connectors/http/internal/security/SecurityContext.java

```diff
@@ -0,0 +1,184 @@
+            Certificate certificate = certificateFactory.generateCertificate(certInputStream);
+            this.keystore.setCertificateEntry(UUID.randomUUID().toString(), certificate);
+            log.info("Certificated added to keyStore ass trusted - " + certPath);
+        } catch (Exception e) {
+            throw new RuntimeException(
+                "Unable to add certificate as trusted to Http Connector security context - "
+                    + certPath,
+                e
+            );
+        }
+    }
+
+    /**
+     * Add certificate and private key that should be used by anny Http Connector instance that uses
+     * this {@link SSLContext} instance. Certificate and key are added only to this Context's
+     * {@link KeyStore} and not for entire JVM.
+     * @param publicKeyPath path to public key/certificate used for mTLS.
+     * @param privateKeyPath path to private key used for mTLS.
+     */
+    public void addMTlsCerts(String publicKeyPath, String privateKeyPath) {
+
+        try {
+            byte[] publicData = Files.readAllBytes(Path.of(publicKeyPath));
+            byte[] privateData = Files.readAllBytes(Path.of(privateKeyPath));
+            byte[] decodedPrivateData = decodePrivateData(privateKeyPath, privateData);
+
+            CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+            Collection<? extends Certificate> chain = certificateFactory.generateCertificates(
+                new ByteArrayInputStream(publicData));
+
+            Key key = KeyFactory.getInstance("RSA")
+                .generatePrivate(new PKCS8EncodedKeySpec(decodedPrivateData));
+
+            this.keystore.setKeyEntry(
+                UUID.randomUUID().toString(),
+                key,
+                this.storePasswordCharArr,
+                chain.toArray(new Certificate[0])
+            );
+        } catch (Exception e) {
+            throw new RuntimeException(
+                "Unable to add client private key/public certificate to Http Connector KeyStore. "
+                    + String.join(",", privateKeyPath, publicKeyPath),
+                e
+            );
+        }
+    }
+
+    /**
+     * Reads private key data. Key can be in PEM and DER coding and in PKCS8 format.
+     * @param privateKeyPath path to private key.
+     * @param privateData read bytes from private key,
+     * @return decoded key data.
+     */
+    private byte[] decodePrivateData(String privateKeyPath, byte[] privateData) {
+
+        // private key must be in PKCS8 format, pem or der.
+        // openssl pkcs8 -topk8 -inform PEM -outform PEM -in client.pem
+        // -out clientPrivateKey.pem -nocrypt
+        if (privateKeyPath.endsWith(".pem")) {
+            String privateString = new String(privateData, Charset.defaultCharset())
+                .replace("-----BEGIN PRIVATE KEY-----", "")
+                .replaceAll(System.lineSeparator(), "")
+                .replaceAll("\\n", "")
```

Comment:
According to https://stackoverflow.com/questions/454908/split-java-string-by-new-line/31060125#31060125 we can use `\R` as a newline regex. And as one of the comments in other answers on that page points out, mixing `System.lineSeparator()` with `\\n` might not be enough: what about a file created on Windows (so with `\r\n` separator) but opened on Linux? Those two lines would match only `\n` and leave `\r` untouched.

Reply:
Done
changed to \R. Thanks for this.

Reply:
I liked the SO comment:

"Yes, it's the best answer. Unfortunate that the question was asked six years too early for this answer."